test/files/pos/InferOverloadedPartialFunction.scala

```diff
@@ -0,0 +1,32 @@
+class MySeq[T] {
+  def map[U](f: T => U): MySeq[U] = new MySeq[U]
+  def map2[U](f: T => U): MySeq[U] = new MySeq[U]
+}
+
+class MyMap[A, B] extends MySeq[(A, B)] {
+  def map[C](f: (A, B) => C): MySeq[C] = new MySeq[C]
+  def map[C, D](f: (A, B) => (C, D)): MyMap[C, D] = new MyMap[C, D]
+
+  def foo(f: Function2[Int, Int, Int]): Unit = ()
+  def foo[R](pf: PartialFunction[(A, B), R]): MySeq[R] = new MySeq[R]
+}
+
+object Test {
+  val m = new MyMap[Int, String]
+
+  // This one already worked because it is not overloaded:
+  m.map2 { case (k, v) => k - 1 }
+
+  // These already worked because preSelectOverloaded eliminated the non-applicable overload:
+  m.map(t => t._1) //
+  m.map((kInFunction, vInFunction) => kInFunction - 1)
+
+  // These worked because the argument types are known for overload resolution:
+  m.map2({ case (k, v) => k - 1 }: PartialFunction[(Int, String), Int])
+  m.map({ case (k, v) => k - 1 }: PartialFunction[(Int, String), Int])
+
+  // These ones did not work before:
+  m.map { case (k, v) => k }
+  m.map { case (k, v) => (k - 1, v) }
```

Comment:
It's not (because the definitions of `map` are wrong). I'll change the example to make this work properly.